src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt

```diff
@@ -138,6 +139,9 @@ class ManagedIndexCoordinator(
             indexStateManagementEnabled = it
             if (!indexStateManagementEnabled) disable() else enable()
         }
+        clusterService.clusterSettings.addSettingsUpdateConsumer(METADATA_SERVICE_ENABLED) {
```

Comment:
Seems like it keeps scheduling in the other functions even if disabled?

Reply:
yeah, add a metadataServiceEnabled variable to save this setting and use it to return early if it's false